storage/internal/benchmarks/continuous_reads.go

```diff
@@ -0,0 +1,202 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"math/rand"
+	"os"
+	"path"
+	"slices"
+	"time"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+	"golang.org/x/sync/errgroup"
+)
+
+// continuousReads is a benchmark that continuously downloads the same set of
+// objects from GCS until the timeout is reached.
+//
+// SET UP
+//   - Initialize structs to populate with the benchmark results.
+//   - Select a random API to use to upload and download the object, unless an
+//     API is set in the command-line,
+//   - Select a random size for the objects; this size will
+//     be between two values configured in the command-line.
+//   - Create a directory of objects of that size in GCS, uploading random
+//     contents directly from memory. The number of objects created is equal to
+//     the command-line configuration for the number of objects in a directory.
+//
+// BENCHMARK
+//   - Grab a storage client from the pool.
+//   - Check if the timeout is exceeded; if so, compile p50, p90 and p99 and return.
+//   - Queue a goroutine that performs a download. This goroutine will run as
+//     soon as there is an available worker. The number of workers that run
+//     concurrently is set in the command-line.
+//   - This goroutine chooses the object to download by receiving an object name
+//     from the channel. At the end of the download, it will return that name to
+//     the channel. That way, no goroutines are reading from the same object at
+//     the same time.
+//   - The time taken is saved in a slice that is used to compile the percentiles
+//     once the timeout is exceeded.
+type continuousReads struct {
+	opts          *benchmarkOptions
+	bucketName    string
+	directoryPath string
+	numWorkers    int
+	objects       chan string // list of objects to synchronize reads
+	results       []time.Duration
+	api           benchmarkAPI
+	objectSize    int64 // every object must be of the same size
+}
+
+func (r *continuousReads) setup(ctx context.Context) error {
+	// Check for context cancellation
+	select {
+	case <-ctx.Done():
+		return ctx.Err()
+	default:
+	}
+
+	// Select API first as it will be the same for all writes/reads
+	api := opts.api
+	if api == mixedAPIs {
+		switch rand.Intn(4) {
+		case 0:
+			api = xmlAPI
+		case 1:
+			api = jsonAPI
+		case 2:
+			api = grpcAPI
+		case 3:
+			api = directPath
+		}
+	}
+	r.api = api
+
+	// Select object size
+	objectSize := opts.objectSize
+	if objectSize == 0 {
+		objectSize = randomInt64(opts.minObjectSize, opts.maxObjectSize)
+	}
+	r.objectSize = objectSize
+
+	// Make a temp dir for this run
+	dir, err := os.MkdirTemp("", "benchmark-experiment-")
```

Comment:
I thought we had discussed downloading to `io.Discard` for now? Otherwise I think we'd run into disk issues.

Reply:
See my other comment.